### PR TITLE
update qsharp to 1.0.33

### PR DESCRIPTION
**Pull request by cqc-melf**
# Description

Please summarise the changes.

# Related issues

Please mention any github issues addressed by this PR.

# Checklist

- [x] I have performed a self-review of my code.
- [ ] I have commented hard-to-understand parts of my code.
- [ ] I have made corresponding changes to the public API documentation.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have updated the changelog with any user-facing changes.

### CODE REVIEW

**Comment by cqc-melf** [comment]
My understanding is that it should be possible to remove the dotnet dependency after this, but it doesn't work yet. I think this misses the releases of additional packages. Happy to do this now and the clean up later, or both later together.

File: setup.py [hunk 1]
```diff
@@ -45,7 +45,7 @@
     include_package_data=True,
     install_requires=[
         "pytket ~= 1.23",
-        "qsharp ~= 0.28.291394",
+        "qsharp ~= 1.0.33",
```
Comment (cqc-alec) [inline setup.py:48]:
```suggestion
        "qsharp ~= 1.0",
```
Further 1.x releases shouldn't break compatibility.

File: docs/changelog.rst [hunk 2]
```diff
@@ -1,6 +1,11 @@
 Changelog
 ~~~~~~~~~
 
+unreleased
+----------
+
+* Updated qsharp version requirement to 1.0.33.
```
Comment (cqc-alec) [inline docs/changelog.rst:7]:
```suggestion
* Updated qsharp version requirement to 1.0.
```

**Comment by cqc-alec** [comment]
> My understanding is that it should be possible to remove the dotnet dependency after this, but it doesn't work yet. I think this misses the releases of additional packages. Happy to do this now and the clean up later, or both later together.

We can do that later. Could you make an issue?

**Review by cqc-alec — CHANGES_REQUESTED** [review]

**Comment by cqc-melf** [comment]
> > My understanding is that it should be possible to remove the dotnet dependency after this, but it doesn't work yet. I think this misses the releases of additional packages. Happy to do this now and the clean up later, or both later together.
> 
> We can do that later. Could you make an issue?

I have created an issue at https://github.com/CQCL/pytket-qsharp/issues/76 and will close this

**Pull request closed without merging**